python/pyspark/sql/context.py

```diff
@@ -72,8 +72,13 @@ def __init__(self, sparkContext, sparkSession=None, jsqlContext=None):
         self._sc = sparkContext
         self._jsc = self._sc._jsc
         self._jvm = self._sc._jvm
+
         if sparkSession is None:
-            sparkSession = SparkSession(sparkContext)
+            if sparkContext is SparkContext._active_spark_context:
+                sparkSession = SparkSession.builder.getOrCreate()
```

Comment:
Thats a good question, lets see what @andrewor14 thinks of changing the builder interface for `SparkSession`